docs/site/migration/models/relations.md

```diff
@@ -6,8 +6,66 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/migration-models-relations.html
 ---
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3948](https://github.com/strongloop/loopback-next/issues/3948).
-" %}
+When you define a relation in a LoopBack 3 model JSON file, the framework will
+create the following artifacts for you automatically:
+
+- Relation metadata defining the target model, foreign key column/property, and
+  so on.
+- Repository-like methods for accessing related model instance(s), for example
+  `Category.prototype.products`.
+- An inclusion resolver to allow clients to request relation traversal in
+  queries and include related models, e.g.
+  `Product.find({include: ['category']})`
+- REST API endpoints for querying a modifying models on the other side of the
+  relation, e.g. `GET /api/categories/1/products`.
+
+In LoopBack 4, these building blocks are typically provided by the application
+developer.
+
+- Relation metadata is defined via model decorators like `@hasMany`.
+- Relation repositories implement APIs for accessing and modifying data of
+  related models.
+- Inclusion resolvers implement relation traversal in queries.
+- Code-generated relation controllers implement REST APIs for model relations.
+
+At the moment, all of these artifacts must be created manually by the
+application developer. This gives developers a lot of power and flexibility in
+customizing the default behavior offered by the framework.
+
+In the future, we would like to provide declarative approach for building model
+relations, where the developer defines relation metadata and the framework
+builds all required artifacts at runtime, similarly to how LoopBack 3 works. You
+can join the discussion in the GitHub issue
+[loopback-next#2483](https://github.com/strongloop/loopback-next/issues/2483).
+
+## Migration path
+
+Follow these steps to migrate a model relation from LoopBack 3 to LoopBack 4:
+
+1. Run `lb4 relation` to define the model relation in your model class, generate
+   code for the relation repository and optionally register inclusion resolver.
+
+2. TBD, `lb4 relation` does not work for me!
```

Comment:
I created two models using `lb model` commands: `Category` and `Product` , and used `lb relation` to configure hasMany/belongsTo relations. After importing those two models to my LB4 app, I run `lb4 relation` to setup hasMany relation in LB4 and received the following error:

```
$ lb4 relation
? Please select the relation type hasMany
? Please select source model Category
? Please select target model Product
? Allow Category queries to include data from related Product instances? Yes
Generation is aborted: Error: Source model primary key does not exist.
```

Here is the LB4 model file:

```ts
@model({
  settings: {strict: false, replaceOnPUT: true, validateUpsert: true, idInjection: true}
})
export class Category extends Entity {
  @property({
    type: 'number',
    id: 1,
    generated: true,
    updateOnly: true,
  })
  id?: number;

  @property({
    type: 'string',
    required: true,
  })
  name: string;

  // Define well-known properties here

  // Indexer property to allow additional data
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [prop: string]: any;

  constructor(data?: Partial<Category>) {
    super(data);
  }
}
```

Maybe the code parsing TypeScript source does not recognize `id: 1` and expects `id: true` instead? I'll investigate.

Reply:
I managed to track down the problem and opened a PR to fix it - see https://github.com/strongloop/loopback-next/pull/4098